File: Codeforces/CF942/d1.py
from math import gcd

def precompute_gcd(n, m):
    gcd_values = [[0] * (m + 1) for _ in range(n + 1)]
    for a in range(1, n + 1):
        for b in range(1, m + 1):
            gcd_values[a][b] = gcd(a, b)
    return gcd_values

def count_ordered_pairs(n, m, gcd_values):
    count = 0
    for a in range(1, n + 1):
        for b in range(1, m + 1):
            if (a + b) % (b * gcd_values[a][b]) == 0:
                count += 1
    return count


t = int(input())


for _ in range(t):
    
    n, m = map(int, input().split())
    gcd_values = precompute_gcd(n,m)

    
    
    print(count_ordered_pairs(n, m,gcd_values))

File: Codeforces/CF942/d1.cpp
#include <iostream>
#include <algorithm>
#include <fstream>
#include<cmath>
using namespace std;

int countOrderedPairs(int n, int m) {
    int count = 0;
    for (int a = 1; a <= n; ++a) {
        for (int b = 1; b <= m; ++b) {
            if ((a + b) % (b * __gcd(a, b)) == 0) {
                count++;
            }
        }
    }
    return count;
}

int main() {
#ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
#endif

    int t;
    cin >> t; // Number of test cases

    while (t--) {
        int n, m;
        cin >> n >> m; // Read n and m for the current test case
        cout << countOrderedPairs(n, m) << endl; // Output the count of ordered pairs
    }

    return 0;
}
